introduce the quickstart mode user guide (#116)
apache/streampipes-website: docs/user-guide-for-quickstart.md

---
id: user-guide-for-quickstart
title: Quickstart Tour
sidebar_label: Tour
---

> StreamPipes is a framework that enables domain experts to model and execute stream processing pipelines in a big data infrastructure.

Reviewer: This is not our value proposition.

> Apache StreamPipes is a self-service Industrial IoT toolbox to enable non-technical users to connect, analyze and explore IoT data streams.

Author: Hi, @bossenti ! I have updated the StreamPipes introduction.

The graphical user interface of StreamPipes is a web application that provides an easy to use solution for domain experts.

Quickstart Mode, ideal for first-time users, allows users to interact with pre-set pipelines, dashboards, and data views to experience StreamPipes' functionalities for IIoT. 
This page provides an overview of StreamPipes' features in Quickstart Mode, guiding users on how to utilize them effectively.

> 

Reviewer: ```suggestion
:::info
Do you have StreamPipes already running in the quickstart mode? If not simply run the command `docker-compose -f docker-compose.quickstart.yml up -d` from the directory `installer/compose`.
:::
```

Author: Hi, @bossenti ! I have added the part as the suggestion, The `info` format looks really cool! Thanks a lot!

> First of all, a user needs to log in to the system. The default login credentials are ``admin@streampipes.apache.org``, password ``admin``

Author: Hi, @bossenti ! I have updated this part in latest commit.

The credentials for the user are specified during the installation process.

## Home
After logging in, the user is redirected on to the home page.
The home page gives an overview over the different features available in the StreamPipes UI.

On the left, the navigation menu can be seen.
You can either use the icons on the left side or click on the menu icon on the
top left to open the details view of the navigation menu.

<div class="my-carousel docs-carousel">
    <img src="/img/quickstart/user-guide/01_home.png" alt="Home"/>
</div>

> ## Pipelines

Reviewer: Can we please start with the Connect module? Simply switch the ordering

Author: Hi, @bossenti ! I have put the `Connect` module as the first module after `Home`.

The _Pipelines_ view provides an overview of all existing pipelines.

A pipeline in Apache StreamPipes describes the transformation process from a data stream to a data sink. Typically, a pipeline consists of at least one data stream (or data set), zero or more data processors and at least one data sink.
Existing pipelines can be managed using this pipeline view.For instance, users can start and stop pipelines or delete them when they are not longer needed.
In the pipeline page, we could see there are 5 pipelines, click 'START ALL PIPELINES' button, we could start all the predefined pipelines.

<div class="my-carousel docs-carousel">
    <img src="/img/quickstart/user-guide/02_pipelines.png" alt="Pipeline Overview"/>
</div>

>  Using the 'Water level trend analysis' pipeline as an example. By clicking the 'Show pipeline' button, we could see the details of the pipeline as the following figure.

Reviewer: Please change order with the image

Reviewer: I was just referring to the fact that the image appears above the statement and not below

Author: Ah, I got it! Thanks for reminding me.

> The pipeline contains 3 major parts:

Reviewer: Would be great if you could provide some rationale _why_ one would like to build such a pipeline. What is the use case for it?

Author: Hi, @bossenti ! Thanks for the suggestion!
 I have added the description of this part in 59 row as follows:
In smart factory , monitoring and analysing machine parameters are essential steps to ensure machine safety, and minimizing downtime. In the industrial machine cooling system or hydraulic machines, water level could be an important part of monitoring parameters to ensure machine's performance and safety.

  - Boolean Counter: to count each 'overflow' status change from 'false' to 'true'. 
  - Numerical Filter: when 'overflow' status change from 'false' to 'true' reaches 5 times, it will trigger Notification.
- **Data Sinks**: 
  - Notification: to push a notification when the measured parameter condition reaches the set condition of the trigger, which are sent to Notification interface. 
  - Data Lake: to store the events in the internal data lake, offering data for dashboard and data explorer.

<div class="my-carousel docs-carousel"> 
<img src="/img/quickstart/user-guide/07_water_level_trend_pipeline.png" alt="Pipeline Detail"/>
</div>

## Connect
The _Connect_ view provides an overview of all existing adapters to let StreamPipes connect data sources.

With StreamPipes Connect it is possible to connect new data sources in StreamPipes with just a few clicks.
It is also possible to connect specific data sources or connect generic sources like message brokers or databases.
If the event schema of the data source is unknown, the system tries to infer the schema by extracting some sample data and analysing it.

> In the Quickstart Mode, we use the Machine Data Simulator and Data stream to simulate the factory machine condition data and environmental data, just click 'START ALL ADAPTERS' button, we could start all existing adapters to offer data source for pipelines.

Reviewer: ```suggestion
In the Quickstart Mode, we use the Machine Data Simulator and Data stream to simulate the factory machine condition data and environmental data, just click the `START ALL ADAPTERS` button. This starts all existing adapters to offer data sources for pipelines.
```

Author: Hi, @bossenti ! have changed the description as suggestion.

<div class="my-carousel docs-carousel">
    <img src="/img/quickstart/user-guide/03_adapters.png" alt="Adapters"/>
</div>

## Dashboard
The _Dashboard_ can be used for multiple use cases.

It is a good way to live monitor a running system in a dashboard, but it can also be used during the pipeline development to get a fast feedback from newly created pipelines.

> Currently 10 different types are available, including line charts, various map visualizations and data tables. The Dashboard feature allows you to select and view real-time data from specific pipelines that interest you.

Reviewer: ```suggestion
Currently 10 different types of visualizations are available, including line charts, various map visualizations and data tables. The Dashboard feature allows you to select and view real-time data from specific pipelines that interest you.
```

Author: Hi, @bossenti ! I have updated the description in this part.
